hyperlink/_url.py

```diff
@@ -970,7 +950,9 @@ def from_text(cls, text):
                             raise URLParseError('port must not be empty')
                         raise URLParseError('expected integer for port, not %r'
                                             % port_str)
-        family, host = parse_host(host)
+        # _, host = parse_host(host)
+        if host:
+            host = host.lstrip('[').rstrip(']')
```

Comment:
After about an hour and a half, I've got an RE and code that mostly does this. It _may_ be better overall, but when I push you'll see how many changes are necessary, especially to keep the error messages friendly (regexes are pretty all-or-nothing).